### ISSUE TITLE
Re: Mathematical Operator Symbols

### ISSUE THREAD

**Issue by benibela** [qt4cg/qtspecs#536]
#460 

`=!>` is not mentioned in A 3.3

Also, I do not think it makes sense to allow mixing both characters in one operator, like in `＜<`. It combines the disadvantages of `<<` and `＜＜` without any advantages

**Comment by dnovatchev** [comment]
@benibela 

Wait for another year and we might see operators such as:

**`==!>>=!->===!>>>=====!!!>>>>>`**

All this while we have a single (proposed) **[fn:chain](https://github.com/qt4cg/qtspecs/issues/517#issuecomment-1561893747)**, that doesn't need any connectors at all.

**Comment by michaelhkay** [comment]
> =!> is not mentioned in A 3.3

I think that's an accident of timing of the two proposals.

> I do not think it makes sense to allow mixing both characters

This is another example of the "paternalism" debate. Just because we don't think it's a good idea for users to do something doesn't necessarily mean that it's a good idea for us to prohibit it. The aim should be to have as few rules and restrictions as possible, not to force our ideas of what constitutes good coding style on users.

On this particular case, though, I don't have strong feelings either way.

**Comment by dnovatchev** [comment]
> This is another example of the "paternalism" debate. Just because we don't think it's a good idea for users to do something doesn't necessarily mean that it's a good idea for us to prohibit it. The aim should be to have as few rules and restrictions as possible, not to force our ideas of what constitutes good coding style on users.
> 
> On this particular case, though, I don't have strong feelings either way.

Says the person who is the only one who possesses the knowledge how to write (a PR and) the text of a spec-document.

Please, distribute this knowledge (the schema and a small walkthrough with examples as a minimum) before requiring non-paternalism from others.

When someone has special rights and privileges bordering on monopolism, who then could this person label as having "paternalistic" opinions?

 What diversity do we really have in this CG, if oly a single person holds the keys to the text of the specification documents?

**Comment by michaelhkay** [comment]
Please check your facts before making such unfounded comments. On my count, nine different people have raised successful pull requests. You are one of the few regular contributors who has made no attempt to do so.

**Comment by dnovatchev** [comment]
> Please check your facts before making such unfounded comments. On my count, nine different people have raised successful pull requests. You are one of the few regular contributors who has made no attempt to do so.

OK then, where can one find any guidance about the schema of the specification document? This needs to be in the most fundamental FAQ for the QT CGs.

**Comment by ChristianGruen** [comment]
> Also, I do not think it makes sense to allow mixing both characters in one operator, like in ＜<. It combines the disadvantages of << and ＜＜ without any advantages

@benibela I agree. It feels more reasonable to me to only allow `＜＜` or `<<`, but not arbitrary combinations.

> OK then, where can one find any guidance about the schema of the specification document?

@dnovatchev Please don't forget I already offered some help.

**Comment by dnovatchev** [comment]
> @dnovatchev Please don't forget I already offered some help.

@ChristianGruen Yes, and I am grateful. I am lucky that I know you.

But what about other people who are not so lucky? Should we rely on luck in order to reverse-engineer how the Specifications are written? I think: No.

Isn't it good to have an agile approach to this project, with goals that are established, agreed upon and understood by everyone, with clear starting documentation and supporting tools,  with probably bi-weekly or monthly Sprints with tasks commitments as part of every Sprint?

We might only need about a week to establish such a process and the needed starting conditions, and soon after that people will notice how much more straightforward our efforts will be.

Right now, at this moment who knows what our goals are for the Release of the Specifications? Have we decided which sets of features must be included for this release? What is the critical path? Deadlines? I am not aware of such agreed-upon and documented decisions.

Without such clarity we still may produce a lot of good work, but it would be much less coordinated and efficient.

The difference in the approach to the specification-design process might explain why some PLs have a major new release almost every year but the last X___ specifications were produced 10 years ago.

This frankly reminds me of the **[Zeno's paradox](https://en.wikipedia.org/wiki/Infinity#Zeno:_Achilles_and_the_tortoise)** 😄

**Comment by ChristianGruen** [comment]
> But what about other people who are not so lucky? 

Let's discuss that in a separate issue.

**Comment by ndw** [comment]
I'm moving the process discussion to email, @dnovatchev . That's not going to be productive in comments on this issue.

**Comment by michaelhkay** [comment]
Change accepted and applied.
